slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -79,10 +79,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
       override def name = super.name.filter(_ != "PRIMARY")
     }
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
+      val BooleanPattern = "^([01])".r
+      val BitPattern = "^b'([01])'".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case (v,"String")    => Some(Some(v))
-        case ("1","Boolean") => Some(Some(true))
-        case ("0","Boolean") => Some(Some(false))
+        case ( BooleanPattern(v) , "Boolean") => Some( Some(if(v eq "1") true else false))
```

Comment:
It should be ```case (BooleanPattern(v), "Boolean") => Some(Some(v == "1"))```. `eq` is reference equally in scala. `if` and `else` there is superfluous.